llvm/test/lab2/mortina_nastya/test.ll

```diff
@@ -0,0 +1,136 @@
+target triple = "x86_64-unknown-linux-gnu"
+
+; Function Attrs: mustprogress noinline nounwind optnone uwtable
+define dso_local void @_Z5Emptyv() #0 {
+entry:
+  ; CHECK: call void @start_instrument()
+  ; CHECK-NEXT: call void @end_instrument()
+  ret void
+}
+
+; Function Attrs: mustprogress noinline nounwind optnone uwtable
+define dso_local noundef i32 @_Z4foo1ii(i32 noundef %a, i32 noundef %b) #0 {
+entry:
+  %a.addr = alloca i32, align 4
+  %b.addr = alloca i32, align 4
+  store i32 %a, ptr %a.addr, align 4
+  store i32 %b, ptr %b.addr, align 4
+  %0 = load i32, ptr %a.addr, align 4
+  %1 = load i32, ptr %b.addr, align 4
+  %add = add nsw i32 %0, %1
+  ret i32 %add
+
+  ; CHECK: call void @start_instrument()
+  ; CHECK-NEXT: %a.addr = alloca i32, align 4
+  ; CHECK-NEXT: %b.addr = alloca i32, align 4
+  ; CHECK-NEXT: store i32 %a, ptr %a.addr, align 4
+  ; CHECK-NEXT: store i32 %b, ptr %b.addr, align 4
+  ; CHECK-NEXT: %0 = load i32, ptr %a.addr, align 4
+  ; CHECK-NEXT: %1 = load i32, ptr %b.addr, align 4
+  ; CHECK-NEXT: %add = add nsw i32 %0, %1
+  ; CHECK-NEXT: call void @end_instrument()
+  ; CHECK-NEXT: ret i32 %add
+}
+
+; Function Attrs: mustprogress noinline nounwind optnone uwtable
+define dso_local noundef i32 @_Z3sumi(i32 noundef %n) #0 {
+entry:
+  %n.addr = alloca i32, align 4
+  %res = alloca i32, align 4
+  %i = alloca i32, align 4
+  store i32 %n, ptr %n.addr, align 4
+  store i32 0, ptr %res, align 4
+  store i32 0, ptr %i, align 4
+  br label %for.cond
+
+  ; CHECK: call void @start_instrument()
+  ; CHECK-NEXT: %n.addr = alloca i32, align 4
+  ; CHECK-NEXT: %res = alloca i32, align 4
+  ; CHECK-NEXT: %i = alloca i32, align 4
+  ; CHECK-NEXT: store i32 %n, ptr %n.addr, align 4
+  ; CHECK-NEXT: store i32 0, ptr %res, align 4
+  ; CHECK-NEXT: store i32 0, ptr %i, align 4
+  ; CHECK-NEXT: br label %for.cond
+
+for.cond:                                         ; preds = %for.inc, %entry
+  %0 = load i32, ptr %i, align 4
+  %1 = load i32, ptr %n.addr, align 4
+  %cmp = icmp slt i32 %0, %1
+  br i1 %cmp, label %for.body, label %for.end
+
+for.body:                                         ; preds = %for.cond
+  %2 = load i32, ptr %i, align 4
+  %mul = mul nsw i32 2, %2
+  %3 = load i32, ptr %res, align 4
+  %add = add nsw i32 %3, %mul
+  store i32 %add, ptr %res, align 4
+  br label %for.inc
+
+for.inc:                                          ; preds = %for.body
+  %4 = load i32, ptr %i, align 4
+  %inc = add nsw i32 %4, 1
+  store i32 %inc, ptr %i, align 4
+  br label %for.cond
+
+for.end:                                          ; preds = %for.cond
+  %5 = load i32, ptr %res, align 4
+  ret i32 %5
+
+  ; CHECK: %5 = load i32, ptr %res, align 4
+  ; CHECK-NEXT: call void @end_instrument()
+  ; CHECK-NEXT: ret i32 %5
+}
+
+; Function Attrs: mustprogress noinline nounwind optnone uwtable
+define dso_local noundef i32 @_Z3cmpii(i32 noundef %a, i32 noundef %b) #0 {
+entry:
+  %retval = alloca i32, align 4
```

Comment:
Please add a test for a function that contains already calls `start_instrument`/`end_instrument`.